coding-guides/starboard.md

```diff
@@ -0,0 +1,199 @@
+# Starboard
+
+This is a long awaited feature, requested by many people.
+
+Let's begin by talking about what a starboard is. This is an example taken from the Discord.js Official Server.
+
+![Starboard](/assets/starboard.png)
+
+> Before we start, there is one thing you need to now. This code is compatible with only Guidebot Class. This is due to a restriction with passing `client` to the messageReactionAdd and messageReactionRemove events.
+
+So, let's begin!
+
+```js
+module.exports = class {
+  constructor(client) {
+    this.client = client;
+  }
+
+async run(reaction, user) {
+  const message = reaction.message;
+  if (reaction.emoji.name !== '⭐') return;
+  if (message.author.id === user.id) return message.channel.send(`${user}, you cannot star your own messages.`);
+  if (message.author.bot) return message.channel.send(`${user}, you cannot star bot messages.`);
+  const { starboardChannel } = this.client.settings.get(message.guild.id);
+  }
+}
+```
+
+This is some simple setup for later on. For ease, I define message as `message`, but this is completely optional. Next, we grab the starboardChannel key from the guilds settings. Then, we preform a couple of checks on the reaction and the message. First, we check if the reaction is the unicode star emote. Then, we preform two checks on the message, which are checking if the user who added the star is the author of the message, and then we check if the author of the message is a bot. If neither of these checks return true, then we're good to go. Let's move on.
+
+```js
```

Comment:
Again, just throwing a codeblock at someone and explaining _afterwards_ what the block does.